docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -98,6 +98,16 @@ error-prone migrations, without downtime, without requiring any
 additional network traffic, and without any extensive communication with
 downstream users.
 
+With SCU, different versions of the _same_ package are now grouped under a common package-name.
```

Comment:
It sounds a bit as if you're saying "these packages are different versions of the same package so they must share a name", while the way I'm thinking about it is: "giving the same name to two packages indicates your intent for them to be considered different versions of the same package". Granted, the difference is subtle and my formulation is awkward, I'm not sure how to best express it.

Reply:
Thanks for this. You've made me realise the importance of the following 2 properties:
- package versions are totally ordered
- package preference maps relate package names to package versions using an injective/1:1 relationship.

Reply:
Refactored text to keep it simple